nossomercadolivre-api/src/main/java/com/github/nossomercadolivre/CategoryDTO.java

```diff
@@ -0,0 +1,62 @@
+package com.github.nossomercadolivre;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.github.nossomercadolivre.validation.UniqueNameCategory;
+
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Size;
+
+import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;
+import static java.util.Objects.isNull;
+
+@JsonInclude(NON_NULL)
+public class CategoryDTO {
+    public Long id;
+
+    @NotBlank
+    @Size(max = 100)
+    @UniqueNameCategory
+    public String name;
+
+    public CategoryDTO categoryMother;
+
+    /**
+     * For frameworks, don't use it
+     */
+    @Deprecated
+    private CategoryDTO() {
+    }
+
+    public CategoryDTO(final Category category) {
+        this.id = category.getId();
+        this.name = category.getName();
+        if (hasObjectCategoryMother()) {
```

Comment:
*Avaliamos de forma lógica o nível de complicação de cada trecho de código. Aqui não tem espaço para feeling. Usamos a teoria da carga cognitiva e medimos o nível de pontos de entendimentos necessário por trecho.*

Toda indireção gera um ponto a mais de entendimento. A ideia é que você coloque este tipo de coisa no sistema quando o objetivo for distribuir a carga cognitiva entre dois ou mais pontos da aplicação. Neste caso, em específico, não distribuimos. Trocamos if != null por if(hasObject...)

Reply:
Concordo e entendo

Porém if != null não é algo que tenha signficado para o negócio, é algo mais longe de uma linguagem ubuquoa e mais perto do "técniquês", perfiro ter métodos que digam qual é a inteção do if.

Vou refatorar e remover o método.


---
Apenas aproveitando para relar, recentemente implementei um método que era algo como: public boolean isEventToCalendar()

No contexto, é um método que visa verificar se o tipo é um evento que pode ser incluido no calendário, e um dev colocou nas condições do if o que faz sentido para a funcionalidade dele, mas que quebrou outras pois a condição dele não é um Evento que poderia ser incluida no Calendário, 

Se ele apenas le-se o nome do método, ele chegaria a conclusão que não poderia ter feito que ele fez (foi oque aconteceu quando o chamei pra ler o código juntos e entender oque estava acontecendo), mas como ele leu apenas o if dentro do método isEventToCalendar pareceu ser uma boa idéia incluir mais uma condição.